### PR TITLE
Update `README.md`

### LINKED ISSUE #85: Update README
The `README` is a little outdated, we should update it.

### PR DESCRIPTION
**Pull request by ehmicky**
Fixes #85.

This updates the `README.md`. It was quite outdated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,106 +6,235 @@
 [![Dependencies](https://david-dm.org/netlify/zip-it-and-ship-it/status.svg)](https://david-dm.org/netlify/zip-it-and-ship-it)
```
Comment (erezrokah) [inline README.md:6]: https://david-dm.org/netlify/zip-it-and-ship-it/status.svg and  https://david-dm.org/netlify/zip-it-and-ship-it are broken

File: README.md [hunk 2]
```diff
@@ -6,106 +6,235 @@
...
 ## Installation
 
 ```bash
 npm install @netlify/zip-it-and-ship-it
 ```
 
-## Usage
+## API
+
+### zipFunctions(srcFolder, destFolder, options?)
```
Comment (erezrokah) [inline README.md:28]: `listFunctions` and `listFunctionsFiles` use h2 headers while `zipFunctions` and `zipFunction` use h3 headers.
We should try to be consistent I think.

File: README.md [hunk 3]
```diff
@@ -6,106 +6,235 @@
...
-/functions/bar/bar.js
-/functions/package.json
-/functions/node_modules/
-```
+- All files/directories within the same directory (except `node_modules`) are included
+- All the `require()`'d files are included
+- All the `require()`'d `node_modules` are included, recursively
+- The following modules are never included:
+  - `@types/*` TypeScript definitions
+  - `aws-sdk`
```
Comment (erezrokah) [inline README.md:61]: Not relevant for the readme, but I think it is best practice to bundle `aws-sdk`:
https://theburningmonk.com/2019/09/should-you-pack-the-aws-sdk-in-your-deployment-artefact/

Comment (ehmicky) [reply README.md:61]: Hum, interesting.

That part of the code was there before I join in, so I am not completely sure why it was there. It looks like it was part of the repo's first commit, created by @biilmann. @biilmann, do you have some additional insights on this?

Comment (erezrokah) [reply README.md:61]: I think the purpose is to minimize the bundle size as NodeJS lambda runtime comes with its own version of `aws-sdk`

File: README.md (hunk 2 above)
Comment (ehmicky) [reply README.md:28]: Right! Fixed.

File: README.md (hunk 1 above)
Comment (ehmicky) [reply README.md:6]: Looks like https://david-dm.org/ is out of business. Removing those links.

**Pull request merged**